pkg/process/checks/container_test.go

```diff
@@ -155,3 +157,62 @@ func TestCalculateCtrPct(t *testing.T) {
 	// prev=-1 because of missing cgroup file in last run
 	assert.Equal(t, float32(-1), calculateCtrPct(3, -1, 0, 0, 1, currentTime, emptyTime))
 }
+
+func TestContainerCheckInit(t *testing.T) {
+	tests := []struct {
+		name                 string
+		override             bool
+		maxPerMessage        int
+		expectedMaxBatchSize int
+	}{
+		{
+			name:                 "default batch size",
+			override:             false,
+			maxPerMessage:        50,
+			expectedMaxBatchSize: ddconfig.DefaultProcessMaxPerMessage,
+		},
+		{
+			name:                 "valid batch size override",
+			override:             true,
+			maxPerMessage:        50,
+			expectedMaxBatchSize: 50,
+		},
+		{
+			name:                 "negative max batch size",
+			override:             true,
+			maxPerMessage:        -1,
+			expectedMaxBatchSize: ddconfig.DefaultProcessMaxPerMessage,
+		},
+		{
+			name:                 "0 max batch size",
+			override:             true,
+			maxPerMessage:        0,
+			expectedMaxBatchSize: ddconfig.DefaultProcessMaxPerMessage,
+		},
+		{
+			name:                 "big max batch size",
+			override:             true,
+			maxPerMessage:        2000,
+			expectedMaxBatchSize: ddconfig.DefaultProcessMaxPerMessage,
+		},
+	}
+
+	assert := assert.New(t)
+	cfg := config.NewDefaultAgentConfig(false)
+	sysInfo := &model.SystemInfo{}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			mockConfig := ddconfig.Mock()
+			if tc.override {
+				mockConfig.Set("process_config.max_per_message", tc.maxPerMessage)
+			}
+
+			Container.Init(cfg, sysInfo)
```

Comment:
Could we instantiate a new container check and call `Init` on it? That way we aren't interfering with other tests.

Same with the RT version

Reply:
I've ended up refactoring a bit the code so this test doesn't exist anymore. But thanks for the feedback :)